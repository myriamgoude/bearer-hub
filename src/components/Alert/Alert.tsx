import * as React from 'react'
import { css } from '@emotion/core'

export const AlertKey = 'alertType'
export enum AlertType {
  Default,
  LoginError
}

interface IAlertProps {
  alertType?: AlertType
  dismissAlert: () => void
}

export class Alert extends React.Component<IAlertProps> {
  private alertMessage(alertType: AlertType) {
    switch (alertType) {
      case AlertType.LoginError:
        return "Something's gone wrong and we couldn't log you in. Please try again."
      default:
        return "Something's gone wrong. Please try again."
    }
  }

  public render() {
    return (
      <div
        css={css`
          position: absolute;
          top: 0;
          right: 0;
          margin: 1em;
          max-width: 25%;
          z-index: 50;
          border-radius: 3px;
          font-color: #dc4c4c;
          background: #fbeded;
          color: red;
          padding: 1em;
          box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.26);
          display: flex;
        `}
      >
        <div
          css={css`
            width: 90%;
          `}
        >
          {this.alertMessage(this.props.alertType || AlertType.Default)}
        </div>
        <div
          css={css`
            padding: 1em;
            cursor: pointer;
            width: 10%;
          `}
          onClick={() => this.props.dismissAlert()}
        >
          X
        </div>
      </div>
    )
  }
}
