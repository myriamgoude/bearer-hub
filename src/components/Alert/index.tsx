export { Alert, AlertType, AlertKey } from './Alert'
