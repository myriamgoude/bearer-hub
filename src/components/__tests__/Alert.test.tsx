import * as React from 'react'
import renderer from 'react-test-renderer'
import { Alert, AlertType } from '../Alert/Alert'

describe('Alert', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Alert alertType={AlertType.LoginError} dismissAlert={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the default correctly', () => {
    const tree = renderer.create(<Alert dismissAlert={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
