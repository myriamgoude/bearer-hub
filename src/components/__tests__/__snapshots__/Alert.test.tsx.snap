// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "v8urg6",
      "next": undefined,
      "styles": "
          position: absolute;
          top: 0;
          right: 0;
          margin: 1em;
          max-width: 25%;
          z-index: 50;
          border-radius: 3px;
          font-color: #dc4c4c;
          background: #fbeded;
          color: red;
          padding: 1em;
          box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.26);
          display: flex;
        ",
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "v5v3zq",
        "next": undefined,
        "styles": "
            width: 90%;
          ",
      }
    }
  >
    Something's gone wrong and we couldn't log you in. Please try again.
  </div>
  <div
    css={
      Object {
        "map": undefined,
        "name": "13y554e",
        "next": undefined,
        "styles": "
            padding: 1em;
            cursor: pointer;
            width: 10%;
          ",
      }
    }
    onClick={[Function]}
  >
    X
  </div>
</div>
`;

exports[`Alert renders the default correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "v8urg6",
      "next": undefined,
      "styles": "
          position: absolute;
          top: 0;
          right: 0;
          margin: 1em;
          max-width: 25%;
          z-index: 50;
          border-radius: 3px;
          font-color: #dc4c4c;
          background: #fbeded;
          color: red;
          padding: 1em;
          box-shadow: 0 2px 4px 0 rgba(3, 13, 54, 0.26);
          display: flex;
        ",
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "v5v3zq",
        "next": undefined,
        "styles": "
            width: 90%;
          ",
      }
    }
  >
    Something's gone wrong. Please try again.
  </div>
  <div
    css={
      Object {
        "map": undefined,
        "name": "13y554e",
        "next": undefined,
        "styles": "
            padding: 1em;
            cursor: pointer;
            width: 10%;
          ",
      }
    }
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
